core/src/main/java/org/elasticsearch/rest/RestRequest.java

```diff
@@ -100,9 +140,53 @@ public final String path() {
 
     public abstract BytesReference content();
 
-    public abstract String header(String name);
+    /**
+     * Get the value of the header or {@code null} if not found. This method only retrieves the first header value if multiple values are
+     * sent. Use of {@link #getAllHeaderValues(String)} should be preferred
+     */
+    public final String header(String name) {
```

Comment:
Yes they were previously implemented in the Netty request. The reasoning for this is for the parsing of the Content-Type header. We cannot expect abstract method calls to work if we call them from the `RestRequest` constructor, since the NettyRequest needs to call this constructor. The other approach I had was lazy initialization in the RestRequest class but man did I hate that approach.